test/lib/yggdrasil/player/registry_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule PlayerRegistryTests do
+  use ExUnit.Case, async: true
+  alias Yggdrasil.Player.Registry
+
+  setup do
+    {:ok, registry} = Registry.start_link
+    {:ok, registry: registry}
+  end
+
+  test "it allows a player to registry as online" do
+    user_id = 1
+    assert :ok = Registry.register_player user_id, self
+  end
+
+  test "it knows if a player is online" do
+    user_id = 1
+    refute Registry.is_online user_id
+    assert :ok = Registry.register_player user_id, self
+    assert Registry.is_online user_id
+  end
+
+  test "it will not allow the same player to register twice" do
+    user_id = 1
+
+    assert :ok = Registry.register_player user_id, self
+
+    register_task = Task.async(fn -> Registry.register_player user_id, self end)
+    assert {:error, :already_registered} = Task.await(register_task)
+  end
+
+  test "it knows when a player exits" do
+    user_id = 1
+
+    register_task = Task.async(fn -> Registry.register_player user_id, self end)
+
+    # verify player was registered successfully
+    assert :ok = Task.await(register_task)
+
+    refute Registry.is_online user_id
```

Comment:
Yeah in this case well one the bug you spotted could the culprit but I do agree that receiving notifications might the most reliable way to ensure that it was done.

I think you can even assert that the test process received a message or something if you went that route.